convert2rhel/pkgmanager/handlers/yum.py

```diff
@@ -0,0 +1,272 @@
+            trying to resolve the dependencies of the transaction. Because of
+            this, we need to loop throught a couple of times until we know that
+            all of the dependencies are resolved without problems.
+
+            You might wonder "why not removing the packages that caused a
+            failure and only loop throught the dep solving again?" Well. Since
+            we are removing the problematic packages using `rpm` and not some
+            specific method in the transaction itself, yum doesn't know that
+            something has changed (The resolveDeps() function doesn't refresh
+            if something else happens outside the transaction), in order to
+            make sure that we won't have any problems with our transaction, it
+            is easier to loop throught everything again and just recreate the
+            transaction, so yum will keep track of what's changed.
+
+            This function should loop max 3 times to get to the point where our
+            transaction doesn't have any problematic packages in there, and the
+            subsequent transactions are "faster" because of some yum internal
+            cache mechanism.
+
+            This might be optimizaed in the future, but for now, it's somewhat
+            reliable.
+
+        :param test_transaction: Determines if the transaction needs to be
+        tested or not.
+        :type test_transaction: bool
+        :raises SystemExit: If we can't resolve the transaction dependencies.
+        :return: A boolean indicating if it was successful or not.
+        :rtype: bool
+        """
+        resolve_deps_finished = False
+
+        # Do not allow this to loop till eternity.
+        attempts = 0
+        while attempts <= MAX_NUM_OF_ATTEMPTS_TO_RESOLVE_DEPS:
+            self._perform_operations()
+            resolved = self._resolve_dependencies(test_transaction)
+
+            if not resolved:
+                loggerinst.info("Retrying to resolve dependencies %s", attempts)
+                attempts += 1
+            else:
+                resolve_deps_finished = True
+                break
+
+        if not resolve_deps_finished:
+            loggerinst.critical("Failed to resolve dependencines in the transaction.")
+
+        if test_transaction:
+            self._base.conf.tsflags.append("test")
+            loggerinst.info("Validating the yum transaction.")
+        else:
+            loggerinst.info("Replacing the system packages.")
+
+        self._process_transaction()
+        # Because we call the same thing multiple times, the rpm database is
+        # not properly closed at the end.
+        # This cause problems because we have another special operation that
+        # happen in the middle of all of this, that is preserving the rhel
+        # kernel. In the YumBase() class there is a special __del__() method
+        # that resolves all of the locks that it places during the transaction.
+        del self._base
```

Comment:
Ah so that's why we don't call `del` in dnf. It should be made clearer in the dnf module

Reply:
We do call `del self._base` in DNF as well, but we need to manually call `self._base.cloes()` to properly cleanup everything at the end of it, and that happens before we call `del self._base`. 

The funny thing about it in DNF is that when we delete the object, the `__del__()` method kicks in, and it calls the `self._bsae.close()` at some point to close the rpmdb and do the cleanup, but for some reason, letting DNF to handle that was not enough, and I had to manually call the `close()` function.  

In any way, I have updated the comment in DNF to be a bit similar to this one.